x-pack/legacy/plugins/siem/public/lib/connectors/servicenow.tsx

```diff
@@ -0,0 +1,222 @@
+  );
+
+  const handleOnChangeMappingConfig = useCallback(
+    (newMapping: CasesConfigurationMapping[]) =>
+      editActionConfig('casesConfiguration', {
+        ...action.config.casesConfiguration,
+        mapping: newMapping,
+      }),
+    [action.config]
+  );
+
+  return (
+    <>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="apiUrl"
+            fullWidth
+            error={errors.apiUrl}
+            isInvalid={isApiUrlInvalid}
+            label={i18n.SERVICENOW_API_URL_LABEL}
+          >
+            <EuiFieldText
+              fullWidth
+              isInvalid={isApiUrlInvalid}
+              name="apiUrl"
+              value={apiUrl}
+              data-test-subj="apiUrlFromInput"
+              placeholder="https://<instance>.service-now.com"
+              onChange={handleOnChangeActionConfig.bind(null, 'apiUrl')}
+              onBlur={handleOnBlurActionConfig.bind(null, 'apiUrl')}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="username"
+            fullWidth
+            error={errors.username}
+            isInvalid={isUsernameInvalid}
+            label={i18n.SERVICENOW_USERNAME_LABEL}
+          >
+            <EuiFieldText
+              fullWidth
+              isInvalid={isUsernameInvalid}
+              name="username"
+              value={username}
+              data-test-subj="usernameFromInput"
+              onChange={handleOnChangeSecretConfig.bind(null, 'username')}
+              onBlur={handleOnBlurSecretConfig.bind(null, 'username')}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiFormRow
+            id="password"
+            fullWidth
+            error={errors.password}
+            isInvalid={isPasswordInvalid}
+            label={i18n.SERVICENOW_PASSWORD_LABEL}
+          >
+            <EuiFieldPassword
+              fullWidth
+              isInvalid={isPasswordInvalid}
+              name="password"
+              value={password}
+              data-test-subj="passwordFromInput"
+              onChange={handleOnChangeSecretConfig.bind(null, 'password')}
+              onBlur={handleOnBlurSecretConfig.bind(null, 'password')}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiSpacer size="xs" />
+          <FieldMapping
+            disabled={false}
+            mapping={mapping as CasesConfigurationMapping[]}
+            onChangeMapping={handleOnChangeMappingConfig}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </>
+  );
+};
+
+const ServiceNowParamsFields: React.FunctionComponent<ActionParamsProps<
+  ServiceNowActionParams
+>> = ({ actionParams, editAction, index, errors }) => {
+  return null;
```

Comment:
huh?

Reply:
This component is being shown when you want to create an action based on an alert (See Kibana's settings). For the moment we do not support it so that's why `null` is returned.